utils/frame/benchmarking-cli/src/writer.rs

```diff
@@ -30,12 +30,15 @@ pub fn open_file(path: &str) -> Result<File, std::io::Error> {
 		.open(path)
 }
 
-pub fn write_trait(file: &mut File, batches: Result<Vec<BenchmarkBatch>, String>) -> Result<(), std::io::Error> {
-	let batches = batches.unwrap();
-
+pub fn write_trait(file: &mut File, batches: Vec<BenchmarkBatch>) -> Result<(), std::io::Error> {
 	let mut current_pallet = Vec::<u8>::new();
 
+	// Skip writing if there are no batches
+	if batches.is_empty() { return Ok(()) }
+
```

Comment:
You can remove this as the loop body is already not executed when `batches`  is empty. Are those batches different from `components`? Because I want output here even for benchmarks without components.

Reply:
Batches are different than components. Batches here means "multiple different pallets". So like if I benchmark `pallet_balances` and `pallet_contracts` in one go, batches would be length 2. That being said, I think this check is still needed, since batches is a vec, and we dont check above this that the vec has some length.

Reply:
> That being said, I think this check is still needed, since batches is a vec, and we dont check above this that the vec has some length.

I do not understand. If `batches` is empty the whole `batches().iter().for_each(...)` statement disappears.

Reply:
But not the stuff at the very end:

```
	// final close trait
	write!(file, "}}\n").unwrap();
```

Idk, maybe this can be written a little smarter?

Reply:
Ahh okay did not saw that. It is fine then.